src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Metadata/SpecialBindingSourceMetadataProvider.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Reflection;
+using System.Threading;
+
+namespace Microsoft.AspNetCore.Mvc.ModelBinding.Metadata
+{
+    public class SpecialBindingSourceMetadataProvider : IBindingMetadataProvider
```

Comment:
Just call this `BindingSourceMetadataProvider`. It's not as special as you think